File: lab1/SimUDuck/RubberDuck.cpp
#include "stdafx.h"
#include "RubberDuck.h"

RubberDuck::RubberDuck()
	:Duck(std::make_unique<FlyNoWay>(), std::make_unique<SqueakBehavior>())
{
}

void RubberDuck::Display() const
{
	std::cout << "I'm rubber duck" << std::endl;
}

File: lw1/src/duck/RubberDuck.php
<?php

class RubberDuck extends Duck
{
    public function __construct()
    {
        parent::__construct(
            new FlyNoWay(),
            new SqueakBehavior(),
            new NoDanceBehavior());
    }

    public function display(): void
    {
        echo 'I\'m rubber duck!' . PHP_EOL;
    }
}